topology/metrics.go

```diff
@@ -232,3 +232,190 @@ func (im *InterfaceMetric) Split(cut int64) (common.Metric, common.Metric) {
 
 	return m1, m2
 }
+
+// BlockdevMetric the blockdevice IO stats
+// easyjson:json
+// gendecoder
+type BlockdevMetric struct {
+	ReadsPerSec        float64 `json:"r/s"`
+	WritesPerSec       float64 `json:"w/s"`
+	ReadsKBPerSec      float64 `json:"rkB/s"`
+	WritesKBPerSec     float64 `json:"wkB/s"`
+	ReadsMergedPerSec  float64 `json:"rrqm/s"`
+	WritesMergedPerSec float64 `json:"wrqm/s"`
+	ReadsMerged        float64 `json:"rrqm"`
+	WritesMerged       float64 `json:"wrqm"`
+	ReadServiceTime    float64 `json:"r_await"`
+	WriteServiceTime   float64 `json:"w_await"`
+	AverageQueueSize   float64 `json:"aqu-sz"`
+	AverageReadSize    float64 `json:"rareq-sz"`
+	AverageWriteSize   float64 `json:"wareq-sz"`
+	ServiceTime        float64 `json:"svctm"`
+	Utilization        float64 `json:"util"`
+	Start              int64   `json:"Start"`
+	Last               int64   `json:"Last"`
+}
+
+// GetBlockdevFields returns the JSON annotation for each field in
+// BlockdevMetric
+func GetBlockdevFields() []string {
```

Comment:
this should be generated by the gendecoder

Reply:
I was thinking the columns displayed in the UI should match the output of "iostat".  So, I call this function in server.go to use these field names rather than what is returned by .GetFieldKeys().

Some of the field names are not allowed in go structures.

Reply:
I not sure that it is good the keep the orignal field name as this is not how the field are reported in skydive in general and can/will raise error with the datastore as not all characters are supported. If we want to keep the original information present in the UI then I would prefer to have a translation there ?

Reply:
Ok - I changed to follow the conventions used for the other probes. :)